airflow/providers/amazon/provider.yaml

```diff
@@ -667,3 +667,6 @@ additional-extras:
   - name: aiobotocore
     dependencies:
       - aiobotocore[boto3]>=2.2.0
+  - name: cncf.kubernetes
+    dependencies:
+      - apache-airflow-providers-cncf-kubernetes>=8.0.0
```

Comment:
I don't think 8.0.0 is right?
There is no such version for cncf.kubernetes

Reply:
I think `>=8.0.0` is fine, and we have to release those three packages together. There should be 8.0.0 release of cncf.kubernetes coming together with releasing of those two providers (the changes in cncf.kubernetes it brings are breaking, so that's expected to be bumped to 8.0.0.

Comment: The "additional" dependencies are pretty special because they are mostly "informatory". Almost no-one will install `apache-airflow-providers-amazon[cncf.kubernetes]==8.3.0` - what is mainly the purpose of those is to be document the dependencies.

Reply:
w8.. I'm missing something here.
In this PR we add a feature to `apache-airflow-providers-cncf-kubernetes` that feature is what needed for amazon and google provider so what we are after is setting dependency for the next version of `apache-airflow-providers-cncf-kubernetes` thus the condition should be:

`apache-airflow-providers-cncf-kubernetes>=7.2.0`

Currently stable version of cncf.kubernetes is 7.1.0

Reply:
> In this PR we add a feature to `apache-airflow-providers-cncf-kubernetes` that feature is what needed for amazon and google provider so what we are after is setting dependency for the next version of `apache-airflow-providers-cncf-kubernetes` thus the condition should be:

Ah ok. If that's not a breaking change, then yes. I have not looked in detail enough and badly classified upcoming cncf.kubernetes version as MAJOR upgrade. 

This is really the result of our release process and why in the past I was leaving all the version upgrades to the release manager "documentation preparation" - because literally we do not know if the next version will be MINOR or MAJOR upgrade until we merge last change. I think it's next to impossible to change it unless we would have a much more active RM role upfront (RM would have to make versioning decisions upfront and for example hold breaking changes from being merged if the next release is non-breaking, but I feel that is quite impossible with our volunteer-driven organisation). 

Ideally we should have in this case `>= NEXT_CNCF_KUBERNETES_PROVIDER_VERSION` that could be automatically changed when the release documentation is prepared - because this is what we have here, really. But it's yet another automation complexity and feels like a bit of an edge-case that can be handled manually. 

And in edge-cases like that it's hard to have a process to follow for every single case like that, so it's case-by-case basis. 

Side note. Assuming that we do not know if the next release is MINOR or MAJOR we could add `>=MINOR_UPGRADE` always. It's a bit limiting and slightly confusing for the users if we do MAJOR release instead, but technically it would work.

For example in this case we could agree to have `>=7.2.0` here independently if the next version is MAJOR or MINOR. Even if we release 8.0.0, that would still technicaly work (8.0.0 >= 7.2.0), but there would be no 7.2.0 release at all., However (except confusing the users) there is also a side effect that it would not alow us (we don't do it anyway but techniclaly there is such a possiblity with mixed-governance model) to release 7.2.0 after the 8.0.0. Technically (and purely theoretically because we've never done so) we **could** have it and then the `>=7.2.0` would be wrong if the features needed will not be cherry-picked to the v7 branch of the provider.

But I think it's better to get it **right** even if it requires a bit of discussion and "humans-in-the-loop" agreeing on details of such versioning question (as we do now).

Reply:
> Side note. Assuming that we do not know if the next release is MINOR or MAJOR we could add >=MINOR_UPGRADE always. It's a bit limiting and slightly confusing for the users if we do MAJOR release instead, but technically it would work.

I like this idea, I added 8.0.0 because I didn't know if there is any breaking changes on the other PRs, but yes 7.2.0 will work for both minor and major releases 👍